src/app/find/mapPage/ResultsBar.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import Icon from '../../../components/icon';
+
+class ResultsBar extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    // this prevents the results bar from updating on map drag
+    if (
+      this.props.filterString === nextProps.filterString &&
+      this.props.isSearching === nextProps.isSearching
+    ) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  noResultsInfo = () => {
+    const { hasResults, filters, clearResults } = this.props;
+    const searchTerm = filters.searchString;
+
+    // No results after searching through the searchbar
+    if (searchTerm && searchTerm.length && !hasResults) {
+      const wordCount = searchTerm.split(' ').length;
+      return (
+        <div className='resultsBarInner'>
+          {wordCount >= 3 ? (
+            <div>
+              Nothing matched for '{searchTerm.trim()}', try using{' '}
+              <span className='font-weight-bold'>fewer search words</span>.
+            </div>
+          ) : (
+            <div>
+              Nothing matched for '{searchTerm.trim()}'.
+              <span className='font-weight-bold'> Try another word.</span>
+            </div>
+          )}
+          <div onClick={clearResults} className='iconWrapper'>
+            <Icon name='times' />
+          </div>
+        </div>
+      );
+    } else if (
+      filters.advancedFilters &&
+      Object.keys(filters.advancedFilters) &&
+      !hasResults
+    ) {
+      // Handle no results after using filters
+      return (
+        <div>
+          Nothing matched, try using{' '}
+          <span className='font-weight-bold'>different filters.</span>
+        </div>
+      );
+    } else {
+      // All other cases
+      // For example, user gets no results after going through clothing eligibility flow.
+      return (
+        <div className='resultsBarInner'>
+          <div>
+            Nothing matched. Try a{' '}
+            <span className='font-weight-bold'>more general search.</span>
+          </div>
+          <div onClick={clearResults} className='iconWrapper'>
+            <Icon name='times' />
+          </div>
+        </div>
+      );
+    }
+  };
+
+  displayContent = () => {
+    const { isSearching, hasResults, filterString } = this.props;
+
+    if (isSearching) {
+      // loading state
```

Comment:
IMO when the line is `return <div>Loading results...</div>;`, then "loading state" is pretty self-evident 😛.
(Even truer for `hasResults` -> `// results found`.)

This is actually great - it means your code and naming choices are readable enough to be self-documenting, which is what all code should strive for.
This is why nowadays a lot of teams actively discourage comments in the code, as well-written code (like this) shouldn't need them in the vast majority of cases.